documentation/service_onboarding.md

```diff
@@ -0,0 +1,164 @@
+	resp, err := client.Create()
+	assert.Nil(t, err)
+	assert.Equal(t, respo.TableResponse.TableName, "tableName")
+}
+```
+
+The first part of the test above is for getting the secrets needed for authentication from your environment, the current practice is to store your test secrets in environment variables.
+
+The rest of the snippet shows a test that creates a single table and asserts that the response from the service has the same table name as the supplied parameter. Every test in Go has to have exactly one parameter, the `t *testing.T` object, and it must begin with `Test`. After making a service call or creating an object you can make assertions on that object by using the external `testify/assert` library. In the example above, we assert that the error returned is `nil`, meaning the call was successful and then we assert that the response object has the same table name as supplied.
+
+You can also use the `testify/require` library instead of `testify/assert` if you want your test to fail as soon as you have an unexpected result.
+
+Check out the docs for more information about the methods available in the [`assert`](https://pkg.go.dev/github.com/stretchr/testify/assert) and [`require`](https://pkg.go.dev/github.com/stretchr/testify/require) libraries.
+
+## Running Tests
+
+The tests are all run using the Go toolchain. To run all your tests simply type `go test` into your command line and you should see an output like:
+```
+PASS
+ok      github.com/Azure/azure-sdk-for-go/sdk/<your_package>        <runtime>s
+```
+
+If you want more details about each of the tests that was run add the `-v` flag for a more verbose output.
+
+### Details about recordings
+
+To be added later.
+
+## Create Pipelines
+
+When you create the first PR for your library you will want to create this PR against a `track2-<package>` library. Submitting PRs to the `main` branch is only for libraries that are to be released. Treating `track2-<package>` as your main development branch will allow nightly CI and live pipeline runs to pick up issues as soon as they are introduced. After creating this PR add a comment with the following:
+```
+/azp run prepare-pipelines
+```
+This creates the pipelines that will verify future PRs. The `azure-sdk-for-go` is tested against versions 1.13 and 1.14 on Windows and Linux. All of your future PRs (regardless of whether they are made to `track2-<package>` or another branch) will be tested against these branches. For more information about individual steps that run in the CI pipelines refer to the [documentation here](./eng_sys.md).
+
+
+## Advance
+For more information about options for configuring your tests and more examples check out the [advanced guide][advanced]
+
+<!-- LINKS -->
+[workspace_setup]: https://www.digitalocean.com/community/tutorials/how-to-install-go-and-set-up-a-local-programming-environment-on-windows-10
+[golang_context]: https://golang.org/pkg/context/#Context
+[advanced]: https://microsoft.com
```

Comment:
That is advanced. What were you thinking of pointing to in the future?

Reply:
The first topic I had in mind was parameterizing tests. There's some nifty ways that can be done in Go and could be very useful for partner teams.